src/Symfony/Bundle/FrameworkBundle/Command/BuildDebugContainerTrait.php

```diff
@@ -60,16 +57,10 @@ protected function getContainerBuilder(KernelInterface $kernel): ContainerBuilde
             }, $kernel, $kernel::class);
             $container = $buildContainer();
 
-            if (str_ends_with($file, '.xml') && is_file(substr_replace($file, '.ser', -4))) {
-                $dumpedContainer = unserialize(file_get_contents(substr_replace($file, '.ser', -4)));
-                $container->setDefinitions($dumpedContainer->getDefinitions());
-                $container->setAliases($dumpedContainer->getAliases());
-                $container->__construct($dumpedContainer->getParameterBag());
-            } else {
-                (new XmlFileLoader($container, new FileLocator()))->load($file);
-                $locatorPass = new ServiceLocatorTagPass();
-                $locatorPass->process($container);
-            }
+            $dumpedContainer = unserialize(file_get_contents(substr_replace($file, '.ser', -4)));
```

Comment:
I suggest changing the value of `debug.container.dump` to have a file name ending in `.ser` directly instead of keeping a `.xml` path while the XML file won't exist.

Reply:
I'd keep the param as is: the xml file still exists and is dumped e.g. for IDE plugins